test/packages/caver.wallet.keyring.js

```diff
@@ -0,0 +1,2075 @@
+/*
+    Copyright 2020 The caver-js Authors
+    This file is part of the caver-js library.
+
+    The caver-js library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const _ = require('lodash')
+const chai = require('chai')
+const sinon = require('sinon')
+const sinonChai = require('sinon-chai')
+
+chai.use(sinonChai)
+
+const expect = chai.expect
+
+const testRPCURL = require('../testrpc')
+
+const Caver = require('../../index.js')
+const utils = require('../../packages/caver-utils')
+const Keyring = require('../../packages/caver-wallet/src/keyring/keyring')
+const Account = require('../../packages/caver-account')
+const AccountKeyPublic = require('../../packages/caver-account/src/accountKey/accountKeyPublic')
+const AccountKeyWeightedMultiSig = require('../../packages/caver-account/src/accountKey/accountKeyWeightedMultiSig')
+const AccountKeyRoleBased = require('../../packages/caver-account/src/accountKey/accountKeyRoleBased')
+const PrivateKey = require('../../packages/caver-wallet/src/keyring/privateKey')
+
+let caver
+
+beforeEach(() => {
+    caver = new Caver(testRPCURL)
+})
+
+function isKeyring(data, { expectedAddress, expectedKey } = {}) {
```

Comment:
What about changing the function name as `validateKeyring`?

Reply:
Oh, i forgot to change ;-;! sorry! i edited function name :)